### ISSUE TITLE
Prevent weights download during inference/testing

### ISSUE THREAD

**Issue by talhaanwarch** [awslabs/autogluon#1111]
I have a trained a vision model and saved it as 
`predictor.save('predictor.ag')`
when I do inference with it , I load the model back again
`predictor = ImagePredictor.load('predictor.ag')`
It start downloading pre-trained weights as 
```
Downloading /root/.mxnet/models/resnet50_v1b-0ecdba34.zip from https://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/gluon/models/resnet50_v1b-0ecdba34.zip...
```
As we are not training the model, pre-trained weights should not be downloaded.

**Comment by Innixma** [comment]
@zhreshold any idea why this is happening? Can we avoid this?

**Comment by talhaanwarch** [comment]
Look like this issue is with the save option, instead of load. If I do
```
import pickle
with open('predictor.ag', 'rb') as fid:
    obj = pickle.load(fid)
```
It starts downloading pre-trained weights. So I think this behavior is saved while saving model

**Comment by talhaanwarch** [comment]
Error log, you can produce it by turning your net off and loading the model
```
File "D:\tutorials\automl\venv\lib\site-packages\gluoncv\auto\estimators\base_estimator.py", line 370, in __setstate__
    self._init_network()
  File "D:\tutorials\automl\venv\lib\site-packages\gluoncv\auto\estimators\image_classification\image_classification.py", line 305, in _init_network
    self.net = get_model(model_name, **kwargs)
  File "D:\tutorials\automl\venv\lib\site-packages\gluoncv\model_zoo\model_zoo.py", line 409, in get_model
    net = _models[name](**kwargs)
  File "D:\tutorials\automl\venv\lib\site-packages\gluoncv\model_zoo\resnetv1b.py", line 366, in resnet50_v1b
    model.load_parameters(get_model_file('resnet%d_v%db'%(50, 1),
  File "D:\tutorials\automl\venv\lib\site-packages\gluoncv\model_zoo\model_store.py", line 298, in get_model_file
    download(_url_format.format(repo_url=repo_url, file_name=file_name),
  File "D:\tutorials\automl\venv\lib\site-packages\gluoncv\utils\download.py", line 67, in download
    r = requests.get(url, stream=True)
  File "D:\tutorials\automl\venv\lib\site-packages\requests\api.py", line 76, in get
    return request('get', url, params=params, **kwargs)
  File "D:\tutorials\automl\venv\lib\site-packages\requests\api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "D:\tutorials\automl\venv\lib\site-packages\requests\sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "D:\tutorials\automl\venv\lib\site-packages\requests\sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "D:\tutorials\automl\venv\lib\site-packages\requests\adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='apache-mxnet.s3-accelerate.dualstack.amazonaws.com', port=443): Max retries exceeded with url: /gluon/models/resnet50_v1b-0ecdba34.zip (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x00000222FD7DB4F0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))
```

**Comment by Innixma** [comment]
Thanks! I've discussed this with @zhreshold and it appears to be a bug in GluonCV. The plan is to fix in GluonCV and then AutoGluon will update to using the fixed release of GluonCV in AG v0.3

**Comment by zhreshold** [comment]
closed with #1164
